### PR TITLE
Add conjure-java-jersey-jakarta-server

### PR DESCRIPTION
**Pull request by mglazer**
## After this PR

==COMMIT_MSG==
Add conjure-java-jersey-jakarta-server

Point all tests to point to this new server, delete legacy test support
==COMMIT_MSG==

### CODE REVIEW

File: conjure-java-jaxrs-client/build.gradle [hunk 1]
```diff
@@ -22,13 +22,17 @@ dependencies {
 
     implementation 'com.palantir.tritium:tritium-registry'
 
+    // include both the old and the new
+    implementation 'javax.ws.rs:javax.ws.rs-api'
+    implementation 'jakarta.ws.rs:jakarta.ws.rs-api'
```
Comment (carterkozak) [inline conjure-java-jaxrs-client/build.gradle]: Should we include both APIs here? I’d assumed we wanted to include neither such that we would only see javax annotations in classpaths that need it? (Once we’ve managed to decouple it from conjure-lib, anyhow)

Comment (mglazer) [reply conjure-java-jaxrs-client/build.gradle]: Not needed anymore.

File: conjure-java-jaxrs-client/src/main/java/com/palantir/conjure/java/client/jaxrs/feignimpl/TextDelegateDecoder.java [hunk 2]
```diff
@@ -50,7 +49,7 @@ public Object decode(Response response, Type type) throws IOException, FeignExce
         }
         // In the case of multiple content types, or an unknown content type, we'll use the delegate instead.
         if (contentTypes.size() == 1
-                && Iterables.getOnlyElement(contentTypes, "").startsWith(MediaType.TEXT_PLAIN)) {
+                && Iterables.getOnlyElement(contentTypes, "").startsWith("text/plain")) {
```
Comment (mglazer) [inline conjure-java-jaxrs-client/src/main/java/com/palantir/conjure/java/client/jaxrs/feignimpl/TextDelegateDecoder.java:52]: I feel like this is an OK string constant to have...

File: conjure-java-jaxrs-client/src/main/java/com/palantir/conjure/java/client/jaxrs/feignimpl/Java8OptionalAwareContract.java [hunk 3]
```diff
@@ -79,31 +79,29 @@ protected void processMetadata(Class<?> targetType, Method method, MethodMetadat
...
             Class<? extends Expander> emptyExpanderClass,
             Class<? extends Expander> nullExpanderClass) {
-        if (paramAnnotations.contains(HeaderParam.class)) {
+        if (Annotations.HEADER_PARAM.matches(paramAnnotations)) {
             metadata.indexToExpanderClass().put(index, emptyExpanderClass);
-        } else if (paramAnnotations.contains(QueryParam.class)) {
+        } else if (Annotations.QUERY_PARAM.matches(paramAnnotations)) {
             metadata.indexToExpanderClass().put(index, nullExpanderClass);
-        } else if (paramAnnotations.contains(PathParam.class)) {
+        } else if (Annotations.PATH_PARAM.matches(paramAnnotations)) {
```
Comment (carterkozak) [inline conjure-java-jaxrs-client/src/main/java/com/palantir/conjure/java/client/jaxrs/feignimpl/Java8OptionalAwareContract.java:98]: :man_facepalming: thanks for fixing these

File: conjure-java-jersey-jakarta-server/src/main/java/com/palantir/conjure/java/server/jersey/ConjureJerseyFeature.java [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.server.jersey;
+
+import com.fasterxml.jackson.jakarta.rs.cbor.JacksonCBORProvider;
```
Comment (mglazer) [inline conjure-java-jersey-jakarta-server/src/main/java/com/palantir/conjure/java/server/jersey/ConjureJerseyFeature.java:19]: The biggest change in the entire copy paste job.

**Review by carterkozak — APPROVED** [review]
Tested with our service framework internally, verified that tests pass and all lockfile changes are expected. Thanks!

**Pull request merged**